### DISCUSSION TITLE
Realm-admin role missing

### DISCUSSION THREAD

**Discussion by Nickk4** [keycloak/keycloak#22870] [Q&A] [answered]
I want to give another user the realm-admin role but only for a specific realm.
Select realm > Users > Select user > Role mapping > Assign role.
Here there are only two roles available to be assigned: `offline_access` and `uma_authorization`. I would expect to also have `realm-admin` role available here to be assigned.

If I subsequently "filter by clients" there are more roles, including the `realm-admin` role for the client `realm-management`. Not sure what this is but it sound more like a role that transcends the different realms.

In the master realm I have the role `admin` and `default-roles-master`. Based on that I would essentially expect to be able to do anything I want to. I am the only user in the master realm.

Why is there no realm-admin role listed? How can I make another user admin of a specific realm?

KeyCloak v21.0.1

**Comment by ssilvert** [comment, marked as answer]
Go to the user for that realm.
Got to Role mapping tab.
Click on Assign role
Click on Filter by clients
Assign "realm-admin" from the realm-management client.

**Reply by Nickk4** [reply]
Thanks, it worked!
